nucypher/characters/lawful.py

```diff
@@ -595,14 +594,20 @@ def get_treasure_map_from_known_ursulas(self, network_middleware, map_id):
 
         return treasure_map
 
-    def generate_work_orders(self, map_id, *capsules, num_ursulas=None, cache=False):
+    def generate_work_orders(self, map_or_id, *capsules, num_ursulas=None, cache=False):
```

Comment:
Not sure if I like this as `map_or_id`. I think I'd rather see a more explicit param for a treasure_map.

Reply:
How do you imagine that? Ideally, I can see `generate_work_orders_from_tmap` in addition to this method (which would take the id). But that's such a terrible name! Suggestions?

Reply:
Wouldn't like to have both `map_id` and `treasure_map` as keyword arguments though where you ned just one - that's terrible!

Reply:
Anyway. I think, I still like `map_or_id` more than alternatives, but adding a type hint

Reply:
I like the `map_or_id` thing. Anyway, a type hint can't hurt.

Reply:
Anyway, looks like after the rebase it's gonna be the way @tuxxy wants anyway!